sklearn/metrics/_scorer.py

```diff
@@ -149,35 +147,25 @@ def __call__(self, estimator, *args, **kwargs):
                     scores[name] = format_exc()
         return scores
 
-    def _use_cache(self, estimator):
-        """Return True if using a cache is beneficial.
-
-        Caching may be beneficial when one of these conditions holds:
-          - `_ProbaScorer` will be called twice.
-          - `_PredictScorer` will be called twice.
-          - `_ThresholdScorer` will be called twice.
-          - `_ThresholdScorer` and `_PredictScorer` are called and
-             estimator is a regressor.
-          - `_ThresholdScorer` and `_ProbaScorer` are called and
-             estimator does not have a `decision_function` attribute.
-
+    def _use_cache(self):
+        """Return True if using a cache is beneficial, thus when a response method will
+        be called several time.
         """
         if len(self._scorers) == 1:  # Only one scorer
             return False
 
-        counter = Counter([type(v) for v in self._scorers.values()])
-
-        if any(
-            counter[known_type] > 1
-            for known_type in [_PredictScorer, _ProbaScorer, _ThresholdScorer]
-        ):
+        counter = Counter(
```

Comment:
Now we can rely solely on the `response_method` to know if we can use the caching mechanism. The different scorers have been removed.